HW3/Program.cs

```diff
@@ -0,0 +1,122 @@
+﻿using HW3.Exceptions;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace HW3
+{
+    internal class Program
+    {
+        private const string usersFile = "Users.txt";
+
+        private static bool ConsoleLogin()
+        {
+            try
+            {
+                string username = ConsoleReadLineWithTextPrompt("Username:\t");
+
+                string password = ConsoleReadLineWithTextPrompt("Password:\t");
+
+                return VerifyCredentials(username, password);
+            }
+            catch (UsersNotFoundException ex)
+            {
+                Console.WriteLine($"Couldn't load user credentials. Please contact @Kaisinel: {ex.Message}");
```

Comment:
Poor kai, getting so much spam 😆 

Also, I'm not sure I'm a fan of this catch statement.  Consider the logic.  First you catch and print this message.  Then you return false so you have to check that return and you print an even more vague message (oh no something went wrong).  Then you loop and you start the program all over again but there is no way to recover from this so things are just going to go wrong again.

On the other hand, you could make "Couldn't load user credentials. Please contact @Kaisinel" the default error message for `UsersNotFoundException` and then don't catch it anywhere.  The program will crash (as it should) and you don't have to return a boolean here so there is no "oh no something went wrong" and the program is simpler.

Reply:
> The program will crash (as it should)

Well actually the assignment was ```If there is no Users.txt file- the application should not throw `UsersNotFoundException`.```. I thought there should be a specific exception but it should not been thrown. I wanted to inform the user instead of crashing the app.